src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/full_screen_waterfall/index.tsx

```diff
@@ -0,0 +1,142 @@
+import { SpanFlyout } from './span_flyout';
+
+export interface FullScreenWaterfallProps {
+  traceId: string;
+  rangeFrom: string;
+  rangeTo: string;
+  dataView: DocViewRenderProps['dataView'];
+  tracesIndexPattern: string;
+  onCloseFullScreen: () => void;
+}
+
+export const FullScreenWaterfall = ({
+  traceId,
+  rangeFrom,
+  rangeTo,
+  dataView,
+  tracesIndexPattern,
+  onCloseFullScreen,
+}: FullScreenWaterfallProps) => {
+  const { transaction } = useRootTransactionContext();
+  const [spanId, setSpanId] = useState<string | null>(null);
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
+  const [scrollElement, setScrollElement] = useState(null);
+  const overlayMaskRef = useRef(null);
+  let flyout;
+
+  const getParentApi = useCallback(
+    () => ({
+      getSerializedStateForChild: () => ({
+        rawState: {
+          traceId,
+          rangeFrom,
+          rangeTo,
+          serviceName: transaction?.[SERVICE_NAME_FIELD],
+          entryTransactionId: transaction?.[TRANSACTION_ID_FIELD] || transaction?.[SPAN_ID_FIELD],
+          scrollElement,
+          onNodeClick: (item: { id: string }) => {
+            setSpanId(item.id);
+            setIsFlyoutVisible(true);
+          },
+        },
+      }),
+    }),
+    [traceId, rangeFrom, rangeTo, transaction, scrollElement]
+  );
+
+  if (isFlyoutVisible && spanId) {
+    flyout = (
+      <SpanFlyout
+        tracesIndexPattern={tracesIndexPattern}
+        spanId={spanId}
+        dataView={dataView}
+        onCloseFlyout={() => {
+          setIsFlyoutVisible(false);
+        }}
+      />
+    );
+  }
+
+  useEffect(() => {
+    if (overlayMaskRef.current) {
+      setScrollElement(overlayMaskRef.current);
+    }
+  }, []);
+
+  return (
+    <>
+      <EuiOverlayMask
+        maskRef={overlayMaskRef}
+        css={{ paddingBlockEnd: '0 !important', overflowY: 'scroll' }}
+      >
+        <EuiFocusTrap css={{ height: '100%', width: '100%' }}>
+          <EuiPanel hasShadow={false} css={{ minHeight: '100%', width: '100%' }}>
+            <EuiFlexGroup alignItems="center">
+              <EuiFlexItem grow={1}>
+                <EuiTitle size="l">
+                  <h2>
+                    {i18n.translate(
+                      'unifiedDocViewer.observability.traces.fullScreenWaterfall.title',
+                      {
+                        defaultMessage: 'Trace timeline',
+                      }
+                    )}
+                  </h2>
+                </EuiTitle>
+              </EuiFlexItem>
+              <EuiFlexItem grow={1} css={{ alignItems: 'end' }}>
+                <EuiButtonIcon
+                  data-test-subj="unifiedDocViewerObservabilityTracesFullScreenWaterfallExitFullScreenButton"
+                  display="base"
+                  iconSize="m"
+                  iconType="fullScreenExit"
+                  aria-label="fullScreenExit"
```

Comment:
We need a translation key for this `aria-label` explaining what this button does